src/rendering/SceneManager.cc

```diff
@@ -1070,7 +1088,7 @@ rendering::LightPtr SceneManager::CreateLight(Entity _id,
   {
     ignerr << "Light with Id: [" << _id << "] already exists in the scene"
            << std::endl;
-    return rendering::LightPtr();
+    return this->dataPtr->lights.find(_id)->second;
```

Comment:
nit: it looks like `this->dataPtr->lights.find(_id)` is called in the body of the enclosing `if` block. We could cache the resulting iterator from the first call to avoid calling `find` a second time